samples/algorithms/sudoku-grover/Program.cs

```diff
@@ -0,0 +1,474 @@
+                if (good)
+                    Console.WriteLine("quantum result verified correct");
+                Pause();
+            }
+            if (puzzleToRun == "9x9-2" || puzzleToRun == "all") 
+            {
+                // Test 9x9 puzzle with quantum - 2 missing number
+                int[,] puzzle9_2 = {
+                    { 0,7,3, 8,9,4, 5,1,2 },
+                    { 9,0,2, 7,3,5, 4,8,6 },
+                    { 8,4,5, 6,1,2, 9,7,3 },
+                    { 7,9,8, 2,6,1, 3,5,4 },
+                    { 5,2,6, 4,7,3, 8,9,1 },
+                    { 1,3,4, 5,8,9, 2,6,7 },
+                    { 4,6,9, 1,2,8, 7,3,5 },
+                    { 2,8,7, 3,5,6, 1,4,9 },
+                    { 3,5,1, 9,4,7, 6,2,8} };
+                Console.WriteLine("Solving 9x9 with 2 missing numbers using Quantum Computing");
+                QuantumSolve(puzzle9_2, 9, 3, sim);
+                var good = puzzle9_2.Cast<int>().SequenceEqual(answer9.Cast<int>());
+                if (good)
+                    Console.WriteLine("quantum result verified correct");
+                Pause();
+            }
+            if (puzzleToRun == "9x9-64" || puzzleToRun == "all") 
+            {
+                // Test hard 9x9 puzzle with classical and quantum 
+                int[,] puzzle9 = {
+                    { 0,0,0, 0,0,0, 0,1,2 },
+                    { 0,0,0, 0,3,5, 0,0,0 },
+                    { 0,0,0, 6,0,0, 0,7,0 },
+                    { 7,0,0, 0,0,0, 3,0,0 },
+                    { 0,0,0, 4,0,0, 8,0,0 },
+                    { 1,0,0, 0,0,0, 0,0,0 },
+                    { 0,0,0, 1,2,0, 0,0,0 },
+                    { 0,8,0, 0,0,0, 0,4,0 },
+                    { 0,5,0, 0,0,0, 6,0,0 } };
+                int[,] puzzle9_copy = CopyIntArray(puzzle9, 9);
+                Console.WriteLine("Solving 9x9 with 64 missing numbers using classical computing");
+                ShowGrid(puzzle9, 9);
+                SolveSudukoClassic(puzzle9, 9, 3);
+                bool good = puzzle9.Cast<int>().SequenceEqual(answer9.Cast<int>());
+                if (!good)
+                    Console.WriteLine("classical test failed");
+                else
+                {
+                    Console.WriteLine("classical test passed");
+                    ShowGrid(puzzle9, 9);
+                }
+                Pause();
+                Console.WriteLine("Solving 9x9 with 64 missing numbers using Quantum Computing. Cntrl-C to stop");
+                QuantumSolve(puzzle9_copy, 9, 3, sim);
+                good = puzzle9_copy.Cast<int>().SequenceEqual(answer9.Cast<int>());
+                if (good)
+                    Console.WriteLine("quantum result verified correct");
+                Pause();
+            }
+            Console.WriteLine("finished");
+
+        }
+
+        // An Empty square with its row and column
+        public class EmptySquare
+        {
+            public int i, j;
```

Comment:
I'd suggest giving each field more descriptive names.
```suggestion
            public int Row;
            public int Column;
```